engines/asylum/resources/actor.cpp

```diff
@@ -0,0 +1,4090 @@
+			}
+
+			// Play sound
+			getSound()->playSound(resourceId, false, -volume, panning);
+		}
+		break;
+	}
+}
+
+void Actor::addReactionHive(int32 reactionIndex, int32 numberValue01Add) {
+	if (reactionIndex > 16)
+		return;
+
+	uint32 i;
+	for (i = 0; i < 8; i++)
+		if (!_reaction[i])
+			break;
+
+	if (i == 8)
+		return;
+
+	if (!hasMoreReactions(reactionIndex, 0))
+		_reaction[i] = reactionIndex;
+
+	if (numberValue01Add)
+		_numberValue01 += numberValue01Add;
+
+	getSound()->playSound(MAKE_RESOURCE(kResourcePackHive, 0));
+}
+
+void Actor::removeReactionHive(int32 reactionIndex, int32 numberValue01Substract) {
+	if (reactionIndex > 16)
+		return;
+
+	if (numberValue01Substract) {
+		_numberValue01 -= numberValue01Substract;
+		if (_numberValue01 < 0)
+			_numberValue01 = 0;
+	}
+
+	if (!numberValue01Substract || !_numberValue01) {
+
+		uint32 i = 0;
+		for (i = 0; i < 8; i++)
+			if (_reaction[i] == reactionIndex)
+				break;
+
+		if (i == 8)
+			return;
+
+		if (i == 7) {
+			_reaction[7] = 0;
+		} else {
+			memmove(&_reaction[i], &_reaction[i + 1], 32 - 4 * i);
+			_reaction[7] = 0;
+		}
+	}
+}
+
+bool Actor::hasMoreReactions(int32 reactionIndex, int32 testNumberValue01) const {
+	if (reactionIndex > 16)
+		return false;
+
+	uint32 i;
+	for (i = 0; i < 8; i++)
+		if (_reaction[i] == reactionIndex)
+			break;
+
+	if (i == 8)
+		return false;
+
+	if (testNumberValue01)
+		return _numberValue01 >= testNumberValue01;
+
+	return true;
+}
+
+bool Actor::canMoveCheckActors(Common::Point *point, ActorDirection dir) {
+	int32 dist = getAbsoluteDistanceForFrame(dir, (_frameIndex >= _frameCount) ? 2 * _frameCount - (_frameIndex + 1) : _frameIndex);
+
+	int32 x = point->x + deltaPointsArray[dir].x * dist - (_field_948 + 10);
+	int32 y = point->y + deltaPointsArray[dir].y * dist - (_field_94C + 10);
+	int32 x1 = x + 2 * _field_948 + 20;
+	int32 y1 = y + 2 * _field_94C + 20;
+
+	for (int32 i = 0; i < (int32)getWorld()->actors.size(); i++) {
+		if (i == _index)
+			continue;
+
+		Actor *actor = getScene()->getActor(i);
+
+		if (!actor->isOnScreen())
+			continue;
+
+		if (actor->_field_944)
+			continue;
+
+		int32 x2 = actor->getPoint1()->x + actor->getPoint2()->x -    (actor->getField948() + 15);
+		int32 y2 = actor->getPoint1()->y + actor->getPoint2()->y -    (actor->getField94C() + 10);
+		int32 x3 = actor->getPoint1()->x + actor->getPoint2()->x + 2 * actor->getField948() + 15;
+		int32 y3 = actor->getPoint1()->y + actor->getPoint2()->y + 2 * actor->getField94C() + 10;
+
+		if (i == getSharedData()->getPlayerIndex() && getWorld()->chapter != kChapter11) {
+			x2 -= 10;
+			y2 -= 10;
+			x3 += 10;
+			y3 += 10;
+		}
+
+		if (getScene()->rectIntersect(x, y, x1, y1, x2, y2, x3, y3)) {
+			if (i)
+				return false;
+
+			int32 x4 = x2 + 10;
+			int32 y4 = y2 + 10;
+			int32 x5 = x3 - 10;
+			int32 y5 = y3 - 10;
+
+			switch (_direction) {
+			default:
+				break;
+
+			case kDirectionNW:
+				if (x4 >= x)
+					break;
+				// Fallback to next case
```

Comment:
Compiler Warning:
````
engines/asylum/resources/actor.cpp: In member function ‘bool Asylum::Actor::canMoveCheckActors(Common::Point*, Asylum::ActorDirection)’:
engines/asylum/resources/actor.cpp:1947:5: warning: this statement may fall through [-Wimplicit-fallthrough=]
     if (x4 >= x)
     ^~
engines/asylum/resources/actor.cpp:1951:4: note: here
    case kDirectionN:
    ^~~~
````
Again, this looks intentional and thus add // fallthrough comment to surpress the warning.